### PR TITLE
add content to custom_documentation files

### PR DESCRIPTION
**Pull request by ferullo**
## Change Summary
This is a replacement PR for https://github.com/elastic/endpoint-package/pull/334

This PR is the third of four chained PRs. I've broken them up to make them reviewable. The final goal is to document what fields are actually available in each Endpoint document type.

1. (This PR) Add the code to read in new YAML documentation and render it in Markdown (https://github.com/elastic/endpoint-package/pull/389)
2. Add minimal YAML documentation that defines all the document types (https://github.com/elastic/endpoint-package/pull/390)
3. (**This PR**) Update all YAML documentation to
4. Render the YAML into pre-committed Markdown files https://github.com/elastic/endpoint-package/pull/392

Please at least skim all four PRs because seeing them all will help understand this PR better.

## Release Target
**This will ideally be backported into 8.9**

### CODE REVIEW

**Review by parkiino — APPROVED** [review]

File: custom_documentation/endpoint/data_stream/alerts/linux/linux_malicious_behavior_alert.yaml [hunk 1]
```diff
@@ -14,8 +15,223 @@ identification:
 fields:
   endpoint:
   - '@timestamp'
+  - Endpoint.policy.applied.artifacts.global.identifiers.name
```
Comment (jdu2600) [inline custom_documentation/endpoint/data_stream/alerts/linux/linux_malicious_behavior_alert.yaml:18]: Hmm. This flat documentation approach feels unwieldy to me... 
I would prefer to see just the fields unique/core to this document up front by default, with grouped links to any common blocks or promoted fields (but also in a way that enables search-ability).
Are there design/UI/documentation folks that could advise us here?

File: custom_documentation/endpoint/data_stream/api/windows/windows_api.yaml [hunk 2]
```diff
@@ -13,3 +15,95 @@ identification:
...
+  - host.os.full
+  - host.os.kernel
+  - host.os.name
+  - host.os.platform
+  - host.os.type
+  - host.os.version
+  - message
+  - process.Ext.ancestry
+  - process.Ext.api.behaviors
+  - process.Ext.api.name
```
Comment (jdu2600) [inline custom_documentation/endpoint/data_stream/api/windows/windows_api.yaml:68]: We'll probably want a way to partition our API event documentation by API.

File: custom_documentation/endpoint/data_stream/api/windows/windows_api.yaml [hunk 3]
```diff
@@ -1,7 +1,9 @@
 overview:
   name: Windows API
```
Comment (jdu2600) [inline custom_documentation/endpoint/data_stream/api/windows/windows_api.yaml:2]: Note - these events are currently diagnostic, and the fields will change slightly before release.

**Review by jdu2600 — APPROVED** [review]

File: custom_documentation/endpoint/data_stream/alerts/linux/linux_malicious_behavior_alert.yaml (hunk 1 above)
Comment (ferullo) [reply custom_documentation/endpoint/data_stream/alerts/linux/linux_malicious_behavior_alert.yaml:18]: I did consider such an approach but decided to stick with simplicity at first. These YAML files are not the finished product, that's the Markdown files rendered from this YAML which does have a requirement to explicitly list all fields.

File: custom_documentation/endpoint/data_stream/api/windows/windows_api.yaml (hunk 3 above)
Comment (ferullo) [reply custom_documentation/endpoint/data_stream/api/windows/windows_api.yaml:2]: I hadn't realized that. Thanks! I will update this documentation to reflect that.

**Pull request merged**